source/Ship.cpp

```diff
@@ -1284,6 +1284,44 @@ void Ship::Launch(list<shared_ptr<Ship>> &ships)
 	for(Bay &bay : bays)
 		if(bay.ship && !Random::Int(40 + 20 * bay.isFighter))
 		{
+			std::shared_ptr<Ship> &ship = bay.ship;
+
+			//do not launch ships with low shields
+			if( ship->Shields() < .9 && ship->Attributes().Get("shields"))
+			{
+				continue;
+			}
+
+			//rearm fighters as they leave the bay
+			set<const Outfit *> toRefill;
+			for(const auto &it : ship->Weapons())
+				if(it.GetOutfit() && it.GetOutfit()->Ammo())
+					toRefill.insert(it.GetOutfit()->Ammo());
+
+			// This is slower than just calculating the proper number to add, but
+			// that does not matter because this is not so time-consuming anyways.
+			for(const Outfit *outfit : toRefill)
+			{
+				while(ship->Attributes().CanAdd(*outfit))
+				{
+					if(Cargo().Get(outfit))
+						Cargo().Remove(outfit);
+					ship->AddOutfit(outfit, 1);
+				}
+			}
+
+
+			//Refuel the fighter.
+			double toTransfer = std::min(ship->attributes.Get("fuel capacity") - ship->fuel, fuel);
+			fuel -= toTransfer;
+			ship->fuel += toTransfer;
+
+			//if still low or out of fuel don't launch
+			if( ship->Fuel() < 0.25 && ship->attributes.Get("fuel capacity"))
```

Comment:
I think a fighter which exclusively requires fuel for dealing damage should not launch.. but a fighter which can deal damage without using fuel is probably OK to launch with no fuel.